aws-iot-thinggroup/src/main/java/software/amazon/iot/thinggroup/Translator.java

```diff
@@ -53,67 +59,31 @@
-        } else if (e instanceof software.amazon.cloudformation.exceptions.ResourceAlreadyExistsException) {
-            return HandlerErrorCode.AlreadyExists;
-        } else if (e instanceof InvalidRequestException) {
-            return HandlerErrorCode.InvalidRequest;
-        } else if (e instanceof CfnInvalidRequestException) {
-            return HandlerErrorCode.InvalidRequest;
-        } else if (e instanceof InvalidQueryException) {
-            return HandlerErrorCode.InvalidRequest;
-        } else if (e instanceof LimitExceededException) {
-            return HandlerErrorCode.ServiceLimitExceeded;
+            return new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, resourceIdentifier, e);
+        } else if (e instanceof ResourceNotFoundException) {
+            return new CfnNotFoundException(ResourceModel.TYPE_NAME, resourceIdentifier, e);
         } else if (e instanceof UnauthorizedException) {
-            return HandlerErrorCode.AccessDenied;
+            return new CfnAccessDeniedException(e);
         } else if (e instanceof InternalFailureException) {
-            return HandlerErrorCode.ServiceInternalError;
-        } else if (e instanceof InternalException) {
-            return HandlerErrorCode.ServiceInternalError;
+            return new CfnInternalFailureException(e);
         } else if (e instanceof ServiceUnavailableException) {
-            return HandlerErrorCode.ServiceInternalError;
+            return new CfnGeneralServiceException(operationName, e);
+        } else if (e instanceof InvalidRequestException || e instanceof InvalidQueryException) {
+            return new CfnInvalidRequestException(e);
+        } else if (e instanceof LimitExceededException) {
+            return new CfnServiceLimitExceededException(ResourceModel.TYPE_NAME, e.getMessage());
+        } else if (e instanceof ConflictingResourceUpdateException || e instanceof VersionConflictException) {
+            return new CfnResourceConflictException(ResourceModel.TYPE_NAME, resourceIdentifier, e.getMessage(), e);
         } else if (e instanceof ThrottlingException) {
-            return HandlerErrorCode.Throttling;
-        } else if (e instanceof ResourceNotFoundException) {
-            return HandlerErrorCode.NotFound;
-        } else if (e instanceof VersionConflictException) {
-            return HandlerErrorCode.ResourceConflict;
-        } else if (e instanceof ConflictingResourceUpdateException | e instanceof DeleteConflictException) {
-            return HandlerErrorCode.ResourceConflict;
-        } else if (e instanceof IotException && ((IotException) e).statusCode() == 403) {
-            return HandlerErrorCode.AccessDenied;
+            return new CfnThrottlingException(operationName, e);
+        } else if (e.statusCode() == HttpStatusCode.FORBIDDEN) {
+            return new CfnAccessDeniedException(operationName, e);
         } else {
-            logger.log(String.format("Unexpected exception \"%s\", stack trace: %s",
```

Comment:
why don't we log unexpected exceptions anymore?

Reply:
There is no need to log these exceptions as the stack trace is logged by handlers on any exceptional scenario.

Reply:
sure